src/ServiceManagement/StorSimple/Commands.StorSimple/Cmdlets/DeviceDetails/SetAzureStorSimpleDevice.cs

```
@@ -0,0 +1,327 @@
+
+                // Get the current device details.
+                var deviceDetails = StorSimpleClient.GetDeviceDetails(DeviceId);
+                if (deviceDetails == null)
+                {
+                    WriteVerbose(string.Format(Resources.NoDeviceFoundWithGivenIdInResourceMessage, StorSimpleContext.ResourceName, DeviceId));
+                    WriteObject(null);
+                    return;
+                }
+
+                // If the device is being configured for the first time, validate that mandatory params 
+                // for first setup have been provided
+                if (!this.IsDeviceConfigurationCompleteForDevice(deviceDetails) && !ValidParamsForFirstDeviceConfiguration())
+                {
+                    WriteVerbose(Resources.MandatoryParamsMissingForInitialDeviceConfiguration);
+                    WriteObject(null);
+                    return;
+                }
+
+                if (!this.ValidateNetworkConfigs(deviceDetails))
+                {
+                    return;
+                }
+
+                // Update device details.
+                this.UpdateDeviceDetails(deviceDetails);
+                
+                // Make request with updated data
+                WriteVerbose("About to run a task to configure the device for the first time!");
+                var taskStatusInfo = StorSimpleClient.UpdateDeviceDetails(deviceDetails);
+                HandleSyncTaskResponse(taskStatusInfo, "Setup");
+                if (taskStatusInfo.AsyncTaskAggregatedResult == AsyncTaskAggregatedResult.Succeeded)
+                {
+                    var updatedDetails = StorSimpleClient.GetDeviceDetails(DeviceId.ToString());
+                    WriteObject(updatedDetails);
+                    WriteVerbose(string.Format(Resources.StorSimpleDeviceUpdatedSuccessfully, updatedDetails.DeviceProperties.FriendlyName));
+                }
+
+            }
+            catch (Exception exception)
+            {
```

Why is the exception being ignored?